internal/pod_helper.go

```diff
@@ -237,6 +222,73 @@ func GetPublicIPSource(pod *corev1.Pod) (fdbv1beta2.PublicIPSource, error) {
 	return fdbv1beta2.PublicIPSource(source), nil
 }
 
+// GetIPFamilyFromPod returns the IP family from the pod configuration.
+func GetIPFamilyFromPod(pod *corev1.Pod) (*int, error) {
+	if GetImageType(pod) == fdbv1beta2.ImageTypeUnified {
+		currentData, present := pod.Annotations[monitorapi.CurrentConfigurationAnnotation]
+		if !present {
+			return nil, fmt.Errorf("could not read current launcher configruations")
+		}
+
+		currentConfiguration := monitorapi.ProcessConfiguration{}
+		err := json.Unmarshal([]byte(currentData), &currentConfiguration)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse current process configuration: %w", err)
+		}
+
+		for _, argument := range currentConfiguration.Arguments {
+			if argument.ArgumentType != monitorapi.IPListArgumentType {
+				continue
+			}
+
+			return pointer.Int(argument.IPFamily), nil
+		}
+
+		// No IP List setting is defined.
+		return nil, nil
+	}
+
+	for _, container := range pod.Spec.Containers {
+		if container.Name != fdbv1beta2.SidecarContainerName {
+			continue
+		}
+
+		for indexOfArgument, argument := range container.Args {
+			if argument == "--public-ip-family" && indexOfArgument < len(container.Args)-1 {
+				familyString := container.Args[indexOfArgument+1]
+				family, err := strconv.Atoi(familyString)
+				if err == nil {
+					return &family, nil
+				}
+				break
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+// GetIPFamily determines the IP family based on the annotation.
+// TODO (johscheuer): Make use of this method once we did the next release 2.2.0. This will ensure that the operator
+// can add the fdbv1beta2.IPFamilyAnnotation to all pods.
+//func GetIPFamily(pod *corev1.Pod) (*int, error) {
```

Comment:
Is the intention that this function will be used in place of `GetIPFamilyFromPod` once we apply all the necessary annotations? What will happen if a cluster does not get re-reconciled between releases, and is left without the annotations for the IP family?

Reply:
That was the initial intention, I think a better approach is to use the value of the annotation if present and if not fall back to the logic from `GetIPFamilyFromPod`. I make the code adjustments later.